Generate 2026.5.0 release notes (#6622)
OttoWinter/esphomedocs: src/content/docs/changelog/2026.5.0.mdx

- **Checked-state binary sensors** ([#16073](https://github.com/esphome/esphome/pull/16073)) report toggle widget state directly, not just press state.
- **Percentage line points** ([#16209](https://github.com/esphome/esphome/pull/16209)) make resolution-independent line widgets straightforward.
- **Touch coordinates in event lambdas** ([#16272](https://github.com/esphome/esphome/pull/16272)). `on_pressed`, `on_pressing`, and `on_release` now receive an additional `point` parameter with object-relative coordinates.
- **`on_update` trigger and `trigger:` option** ([#16312](https://github.com/esphome/esphome/pull/16312)) distinguish programmatic value changes from user interactions for numbers and sensors.

The standalone `mapping` component also gained default values and metadata for cross-component validation ([#15861](https://github.com/esphome/esphome/pull/15861)).

## Zigbee Expanded to ESP32 H2 and C6

> [@luar123](https://github.com/luar123) brought ESP32-based Zigbee support to ESPHome in [#11553](https://github.com/esphome/esphome/pull/11553), porting the standalone `zigbee_esphome` external component into core. The implementation runs on the radio-equipped ESP32-C6 and ESP32-H2 variants and publishes ESPHome `binary_sensor` entities as Zigbee `binary_input` clusters, recognized automatically by ZHA and zigbee2mqtt. Devices keep their network/binding/reporting configuration across reboots.

Reviewer: ```suggestion
[@luar123](https://github.com/luar123) brought ESP32-based Zigbee support to ESPHome in [#11553](https://github.com/esphome/esphome/pull/11553), porting the standalone `zigbee_esphome` external component into core. The implementation runs on the radio-equipped ESP32-C6 and ESP32-H2 variants and publishes ESPHome `binary_sensor` entities as Zigbee `binary_input` clusters, recognized automatically by ZHA and zigbee2mqtt. Zigbee router and end devices are supported.
```
Add router/end device support.
And remove the last sentence as it is expected behavior. No need to highlight.

Reviewer 2: Thanks @luar123
